### ISSUE TITLE
It cannot work in MacOs

### ISSUE THREAD

**Issue by lwx5924** [kirbs-/hide_code#61]
Hi, after I Install hide_code as following step:

    pip install hide_code
    jupyter nbextension install --py hide_code
    jupyter nbextension enable --py hide_code
    jupyter serverextension enable --py hide_code

I try to export PDF, but it still has "Code" in output.

then I go to
http://localhost:8888/nbextensions?nbextension=hide_code/hide_code.

I found there was a warning below:

-------------- warning start ------------
hide_code/hide_code

This nbextension is enabled in the notebook json config, but doesn't provide a yaml file to tell us how to configure it. You can still enable or disable it from here, though.
section: notebook
require path: hide_code/hide_code
compatibility: ?.x

-------------- warning end ------------

How can I solve this problem?
Thanks very much for your help.

My environment :
jupyter notebook --version                                                                                             
5.4.1

jupyter nbconvert --version                                                                                            
5.3.1

**Comment by elliottmcollins** [comment]
Sadly, it looks like this extension only works on Linux machines. Can anyone confirm?

**Comment by kirbs-** [comment]
It should work fine. I'l take a look.

**Comment by kirbs-** [comment]
Please run `jupyter nbextension list` and `jupyter serverextension list` and post the output.

**Comment by GregSilverman** [comment]
I'm on Mac OS too, getting the same warning as noted above. 

Output for `jupyter nbextension list` is:

`Known nbextensions:
  config dir: /Users/gregsilverman/.jupyter/nbconfig
    notebook section
      jupyter-js-widgets/extension disabled
      beakerx/extension  enabled
      - Validating: OK
      nbextensions_configurator/config_menu/main  enabled
      - Validating: problems found:
        - require?  X nbextensions_configurator/config_menu/main
      contrib_nbextensions_help_item/main  enabled
      - Validating: OK
      collapsible_headings/main  enabled
      - Validating: OK
      printview/main  enabled
      - Validating: OK
      toc2/main  enabled
      - Validating: OK
      codefolding/main  enabled
      - Validating: OK
      hide_input/main  enabled
      - Validating: OK
      hide_code/hide_code  enabled
      - Validating: OK
    tree section
      nbextensions_configurator/tree_tab/main  enabled
      - Validating: problems found:
        - require?  X nbextensions_configurator/tree_tab/main
    edit section
      codefolding/edit  enabled
      - Validating: OK
  config dir: /Users/gregsilverman/anaconda/etc/jupyter/nbconfig
    notebook section
      jupyter-js-widgets/extension disabled
      beakerx/extension  enabled
      - Validating: OK
      hide_code/hide_code  enabled
      - Validating: OK
    tree section
      beakerx/tree  enabled
      - Validating: OK
  config dir: /usr/local/etc/jupyter/nbconfig
    tree section
      ipyparallel/main  enabled
      - Validating: OK`

and `jupyter serverextension list` is

`config dir: /Users/gregsilverman/.jupyter
    jupyter_nbextensions_configurator  enabled
    - Validating...
      jupyter_nbextensions_configurator  OK
    hide_code  enabled
    - Validating...
      hide_code  OK
config dir: /Users/gregsilverman/anaconda/etc/jupyter
    jupyterlab  enabled
    - Validating...
      jupyterlab  OK
    beakerx  enabled
    - Validating...
      beakerx  OK
config dir: /usr/local/etc/jupyter
    ipyparallel.nbextension  enabled
    - Validating...
      ipyparallel.nbextension  OK`

however, my issue is that even with the plugin-enabled, I keep getting a 404 error when trying to output to HTML.

**Comment by kirbs-** [comment]
@GregSilverman please post the console output when you try to export via a notebook.

**Comment by GregSilverman** [comment]
Here it is from when I try to export HTML directly from the notebook (that is, not via the command line)"

```
[I 14:00:21.133 NotebookApp] JupyterLab application directory is /Users/gregsilverman/anaconda/share/jupyter/lab
[I 14:00:21.142 NotebookApp] Serving notebooks from local directory: /Users/gregsilverman
[I 14:00:21.142 NotebookApp] 0 active kernels
[I 14:00:21.142 NotebookApp] The Jupyter Notebook is running at:
[I 14:00:21.142 NotebookApp] http://localhost:8888/?token=10180d1ea82f4aa060e93f006b3fbaa72a63a172a8bccc0e
[I 14:00:21.142 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[C 14:00:21.143 NotebookApp] 
    
    Copy/paste this URL into your browser when you connect for the first time,
    to login with a token:
        http://localhost:8888/?token=10180d1ea82f4aa060e93f006b3fbaa72a63a172a8bccc0e

[I 14:00:21.652 NotebookApp] Accepting one-time-token-authenticated connection from ::1
[I 14:00:37.150 NotebookApp] Kernel started: 291410ce-5d44-409c-ba54-10cb9865319b
[W 14:00:37.552 NotebookApp] 404 GET /nbextensions/widgets/notebook/js/extension.js?v=20180604140020 (::1) 15.50ms referer=http://localhost:8888/notebooks/development/python/mb_analysis/mb_latest.ipynb
[I 14:00:39.897 NotebookApp] Adapting to protocol v5.1 for kernel 291410ce-5d44-409c-ba54-10cb9865319b
[I 14:01:08.551 NotebookApp] Starting buffering for 291410ce-5d44-409c-ba54-10cb9865319b:8a077c519a7147e1bc2667d0235a6c4f
[I 14:01:08.562 NotebookApp] Adapting to protocol v5.1 for kernel 291410ce-5d44-409c-ba54-10cb9865319b
[W 14:01:08.563 NotebookApp] 404 GET /notebooks/development/python/mb_analysis/mb_latest.ipynb/export/html (::1): No such file or directory: development/python/mb_analysis/mb_latest.ipynb/export/html
[W 14:01:08.564 NotebookApp] 404 GET /notebooks/development/python/mb_analysis/mb_latest.ipynb/export/html (::1) 1.24ms referer=http://localhost:8888/notebooks/development/python/mb_analysis/mb_latest.ipynb
[I 14:01:08.564 NotebookApp] Restoring connection for 291410ce-5d44-409c-ba54-10cb9865319b:8a077c519a7147e1bc2667d0235a6c4f
[I 14:01:08.572 NotebookApp] Starting buffering for 291410ce-5d44-409c-ba54-10cb9865319b:8a077c519a7147e1bc2667d0235a6c4f
```

and here is when I try to export as PDF:

```
[I 14:01:08.564 NotebookApp] Restoring connection for 291410ce-5d44-409c-ba54-10cb9865319b:8a077c519a7147e1bc2667d0235a6c4f
[I 14:01:08.572 NotebookApp] Starting buffering for 291410ce-5d44-409c-ba54-10cb9865319b:8a077c519a7147e1bc2667d0235a6c4f
[I 14:03:56.538 NotebookApp] Adapting to protocol v5.1 for kernel 291410ce-5d44-409c-ba54-10cb9865319b
[I 14:03:56.538 NotebookApp] Restoring connection for 291410ce-5d44-409c-ba54-10cb9865319b:8a077c519a7147e1bc2667d0235a6c4f
[I 14:04:03.413 NotebookApp] Starting buffering for 291410ce-5d44-409c-ba54-10cb9865319b:8a077c519a7147e1bc2667d0235a6c4f
[I 14:04:03.423 NotebookApp] Adapting to protocol v5.1 for kernel 291410ce-5d44-409c-ba54-10cb9865319b
[W 14:04:03.424 NotebookApp] 404 GET /notebooks/development/python/mb_analysis/mb_latest.ipynb/export/pdf (::1): No such file or directory: development/python/mb_analysis/mb_latest.ipynb/export/pdf
[W 14:04:03.425 NotebookApp] 404 GET /notebooks/development/python/mb_analysis/mb_latest.ipynb/export/pdf (::1) 1.26ms referer=http://localhost:8888/notebooks/development/python/mb_analysis/mb_latest.ipynb#
[I 14:04:03.426 NotebookApp] Restoring connection for 291410ce-5d44-409c-ba54-10cb9865319b:8a077c519a7147e1bc2667d0235a6c4f
[I 14:04:03.430 NotebookApp] Starting buffering for 291410ce-5d44-409c-ba54-10cb9865319b:8a077c519a7147e1bc2667d0235a6c4f

```

**Comment by elliottmcollins** [comment]
@kirbs- , I got it installed (unrelated pip issue), but when hide_code is enabled, notebooks can't open, returning **500: Internal Server Error** instead. I don't know if this is an OSx issue, but it's definitely an issue.

The console output when opening a notebook is:

`[I 13:15:26.247 NotebookApp] [nb_conda_kernels] enabled, 5 kernels found
[I 13:15:26.385 NotebookApp] hide_code: Attempting to load hid_code export handler extensions.
[I 13:15:26.388 NotebookApp] hide_code: Hide_code export handler extensions loaded.
[I 13:15:26.422 NotebookApp] JupyterLab beta preview extension loaded from /Users/elliottcollins/anaconda/lib/python3.5/site-packages/jupyterlab
[I 13:15:26.422 NotebookApp] JupyterLab application directory is /Users/elliottcollins/anaconda/share/jupyter/lab
[I 13:15:26.482 NotebookApp] [nb_anacondacloud] enabled
[I 13:15:26.486 NotebookApp] [nb_conda] enabled
[I 13:15:26.489 NotebookApp] ✓ nbpresent HTML export ENABLED
[W 13:15:26.489 NotebookApp] ✗ nbpresent PDF export DISABLED: No module named 'nbbrowserpdf'
[I 13:15:26.495 NotebookApp] Serving notebooks from local directory: /Users/elliottcollins/GDrive/analysis/impact_sort
[I 13:15:26.495 NotebookApp] 0 active kernels
[I 13:15:26.495 NotebookApp] The Jupyter Notebook is running at:
[I 13:15:26.496 NotebookApp] http://localhost:8888/?token=d762198d16f0e0cb4bbd704f0acb832e541ea46beafef1f0
[I 13:15:26.496 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[C 13:15:26.497 NotebookApp]

    Copy/paste this URL into your browser when you connect for the first time,
    to login with a token:
        http://localhost:8888/?token=d762198d16f0e0cb4bbd704f0acb832e541ea46beafef1f0&token=d762198d16f0e0cb4bbd704f0acb832e541ea46beafef1f0
[I 13:15:26.653 NotebookApp] Accepting one-time-token-authenticated connection from ::1
[E 13:15:31.692 NotebookApp] Uncaught exception GET /notebooks/Impact%20Sort%20Projections.ipynb (::1)
    HTTPServerRequest(protocol='http', host='localhost:8888', method='GET', uri='/notebooks/Impact%20Sort%20Projections.ipynb', version='HTTP/1.1', remote_ip='::1')
    Traceback (most recent call last):
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/tornado/web.py", line 1541, in _execute
        result = method(*self.path_args, **self.path_kwargs)
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/tornado/web.py", line 2949, in wrapper
        return method(self, *args, **kwargs)
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/notebook/notebook/handlers.py", line 59, in get
        get_custom_frontend_exporters=get_custom_frontend_exporters
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/notebook/base/handlers.py", line 424, in render_template
        return template.render(**ns)
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/jinja2/environment.py", line 1008, in render
        return self.environment.handle_exception(exc_info, True)
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/jinja2/environment.py", line 780, in handle_exception
        reraise(exc_type, exc_value, tb)
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/jinja2/_compat.py", line 37, in reraise
        raise value.with_traceback(tb)
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/notebook/templates/notebook.html", line 1, in top-level template code
        {% extends "page.html" %}
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/notebook/templates/page.html", line 154, in top-level template code
        {% block header %}
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/notebook/templates/notebook.html", line 117, in block "header"
        {% for exporter in get_custom_frontend_exporters() %}
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/notebook/notebook/handlers.py", line 24, in get_custom_frontend_exporters
        exporter = get_exporter(name)()
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/nbconvert/exporters/base.py", line 95, in get_exporter
        return entrypoints.get_single('nbconvert.exporters', name).load()
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/entrypoints.py", line 81, in load
        obj = getattr(obj, attr)
    AttributeError: module 'hide_code' has no attribute 'HideCodeLatexPDFExporter'
[E 13:15:31.727 NotebookApp] {
      "Connection": "keep-alive",
      "Accept-Language": "en-US,en;q=0.9",
      "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8",
      "Referer": "http://localhost:8888/tree",
      "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/66.0.3359.181 Safari/537.36",
      "Host": "localhost:8888",
      "Cookie": "_xsrf=2|b23ab7ab|0b300fb5e6a8bc7867593fd0fd920127|1528143260; username-localhost-8888=\"2|1:0|10:1528143326|23:username-localhost-8888|44:YTU4MzZmMzU1ZjE3NGQ3YTgzYjVmZjBjMjdlYzdjYTE=|636e44c0aebff29c620d2a6461f86e67844173379895a325d628998abb741a29\"",
      "Upgrade-Insecure-Requests": "1",
      "Accept-Encoding": "gzip, deflate, br"
    }
[E 13:15:31.727 NotebookApp] 500 GET /notebooks/Impact%20Sort%20Projections.ipynb (::1) 315.92ms referer=http://localhost:8888/tree
[E 13:15:42.380 NotebookApp] Uncaught exception GET /notebooks/Impact%20Sort.ipynb (::1)
    HTTPServerRequest(protocol='http', host='localhost:8888', method='GET', uri='/notebooks/Impact%20Sort.ipynb', version='HTTP/1.1', remote_ip='::1')
    Traceback (most recent call last):
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/tornado/web.py", line 1541, in _execute
        result = method(*self.path_args, **self.path_kwargs)
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/tornado/web.py", line 2949, in wrapper
        return method(self, *args, **kwargs)
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/notebook/notebook/handlers.py", line 59, in get
        get_custom_frontend_exporters=get_custom_frontend_exporters
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/notebook/base/handlers.py", line 424, in render_template
        return template.render(**ns)
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/jinja2/environment.py", line 1008, in render
        return self.environment.handle_exception(exc_info, True)
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/jinja2/environment.py", line 780, in handle_exception
        reraise(exc_type, exc_value, tb)
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/jinja2/_compat.py", line 37, in reraise
        raise value.with_traceback(tb)
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/notebook/templates/notebook.html", line 1, in top-level template code
        {% extends "page.html" %}
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/notebook/templates/page.html", line 154, in top-level template code
        {% block header %}
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/notebook/templates/notebook.html", line 117, in block "header"
        {% for exporter in get_custom_frontend_exporters() %}
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/notebook/notebook/handlers.py", line 24, in get_custom_frontend_exporters
        exporter = get_exporter(name)()
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/nbconvert/exporters/base.py", line 95, in get_exporter
        return entrypoints.get_single('nbconvert.exporters', name).load()
      File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/entrypoints.py", line 81, in load
        obj = getattr(obj, attr)
    AttributeError: module 'hide_code' has no attribute 'HideCodeLatexPDFExporter'
[E 13:15:42.382 NotebookApp] {
      "Connection": "keep-alive",
      "Accept-Language": "en-US,en;q=0.9",
      "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8",
      "Referer": "http://localhost:8888/tree",
      "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/66.0.3359.181 Safari/537.36",
      "Host": "localhost:8888",
      "Cookie": "_xsrf=2|b23ab7ab|0b300fb5e6a8bc7867593fd0fd920127|1528143260; username-localhost-8888=\"2|1:0|10:1528143326|23:username-localhost-8888|44:YTU4MzZmMzU1ZjE3NGQ3YTgzYjVmZjBjMjdlYzdjYTE=|636e44c0aebff29c620d2a6461f86e67844173379895a325d628998abb741a29\"",
      "Upgrade-Insecure-Requests": "1",
      "Accept-Encoding": "gzip, deflate, br"
    }
[E 13:15:42.383 NotebookApp] 500 GET /notebooks/Impact%20Sort.ipynb (::1) 148.91ms referer=http://localhost:8888/tree`

**Comment by elliottmcollins** [comment]
@kirbs- , I got it installed (unrelated pip issue), but when hide_code is enabled, notebooks can't open, returning 500: Internal Server Error instead. I don't know if this is an OSx issue, but it's definitely an issue.

The console output when opening a notebook is:

> [I 13:15:26.247 NotebookApp] [nb_conda_kernels] enabled, 5 kernels found

> [I 13:15:26.385 NotebookApp] hide_code: Attempting to load hid_code export handler extensions.
> [I 13:15:26.388 NotebookApp] hide_code: Hide_code export handler extensions loaded.
> [I 13:15:26.422 NotebookApp] JupyterLab beta preview extension loaded from /Users/elliottcollins/anaconda/lib/python3.5/site-packages/jupyterlab
> [I 13:15:26.422 NotebookApp] JupyterLab application directory is /Users/elliottcollins/anaconda/share/jupyter/lab
> [I 13:15:26.482 NotebookApp] [nb_anacondacloud] enabled
> [I 13:15:26.486 NotebookApp] [nb_conda] enabled
> [I 13:15:26.489 NotebookApp] ✓ nbpresent HTML export ENABLED
> [W 13:15:26.489 NotebookApp] ✗ nbpresent PDF export DISABLED: No module named 'nbbrowserpdf'
> [I 13:15:26.495 NotebookApp] Serving notebooks from local directory: /Users/elliottcollins/GDrive/analysis/impact_sort
> [I 13:15:26.495 NotebookApp] 0 active kernels
> [I 13:15:26.495 NotebookApp] The Jupyter Notebook is running at:
> [I 13:15:26.496 NotebookApp] http://localhost:8888/?token=d762198d16f0e0cb4bbd704f0acb832e541ea46beafef1f0
> [I 13:15:26.496 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
> [C 13:15:26.497 NotebookApp]
> 
> Copy/paste this URL into your browser when you connect for the first time,
> to login with a token:
>     http://localhost:8888/?token=d762198d16f0e0cb4bbd704f0acb832e541ea46beafef1f0&token=d762198d16f0e0cb4bbd704f0acb832e541ea46beafef1f0
> [I 13:15:26.653 NotebookApp] Accepting one-time-token-authenticated connection from ::1
> [E 13:15:31.692 NotebookApp] Uncaught exception GET /notebooks/Impact%20Sort%20Projections.ipynb (::1)
> HTTPServerRequest(protocol='http', host='localhost:8888', method='GET', uri='/notebooks/Impact%20Sort%20Projections.ipynb', version='HTTP/1.1', remote_ip='::1')
> Traceback (most recent call last):
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/tornado/web.py", line 1541, in _execute
> result = method(self.path_args, **self.path_kwargs)
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/tornado/web.py", line 2949, in wrapper
> return method(self, args, **kwargs)
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/notebook/notebook/handlers.py", line 59, in get
> get_custom_frontend_exporters=get_custom_frontend_exporters
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/notebook/base/handlers.py", line 424, in render_template
> return template.render(**ns)
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/jinja2/environment.py", line 1008, in render
> return self.environment.handle_exception(exc_info, True)
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/jinja2/environment.py", line 780, in handle_exception
> reraise(exc_type, exc_value, tb)
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/jinja2/_compat.py", line 37, in reraise
> raise value.with_traceback(tb)
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/notebook/templates/notebook.html", line 1, in top-level template code
> {% extends "page.html" %}
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/notebook/templates/page.html", line 154, in top-level template code
> {% block header %}
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/notebook/templates/notebook.html", line 117, in block "header"
> {% for exporter in get_custom_frontend_exporters() %}
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/notebook/notebook/handlers.py", line 24, in get_custom_frontend_exporters
> exporter = get_exporter(name)()
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/nbconvert/exporters/base.py", line 95, in get_exporter
> return entrypoints.get_single('nbconvert.exporters', name).load()
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/entrypoints.py", line 81, in load
> obj = getattr(obj, attr)
> AttributeError: module 'hide_code' has no attribute 'HideCodeLatexPDFExporter'
> [E 13:15:31.727 NotebookApp] {
> "Connection": "keep-alive",
> "Accept-Language": "en-US,en;q=0.9",
> "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,/;q=0.8",
> "Referer": "http://localhost:8888/tree",
> "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/66.0.3359.181 Safari/537.36",
> "Host": "localhost:8888",
> "Cookie": "_xsrf=2|b23ab7ab|0b300fb5e6a8bc7867593fd0fd920127|1528143260; username-localhost-8888="2|1:0|10:1528143326|23:username-localhost-8888|44:YTU4MzZmMzU1ZjE3NGQ3YTgzYjVmZjBjMjdlYzdjYTE=|636e44c0aebff29c620d2a6461f86e67844173379895a325d628998abb741a29"",
> "Upgrade-Insecure-Requests": "1",
> "Accept-Encoding": "gzip, deflate, br"
> }
> [E 13:15:31.727 NotebookApp] 500 GET /notebooks/Impact%20Sort%20Projections.ipynb (::1) 315.92ms referer=http://localhost:8888/tree
> [E 13:15:42.380 NotebookApp] Uncaught exception GET /notebooks/Impact%20Sort.ipynb (::1)
> HTTPServerRequest(protocol='http', host='localhost:8888', method='GET', uri='/notebooks/Impact%20Sort.ipynb', version='HTTP/1.1', remote_ip='::1')
> Traceback (most recent call last):
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/tornado/web.py", line 1541, in _execute
> result = method(self.path_args, **self.path_kwargs)
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/tornado/web.py", line 2949, in wrapper
> return method(self, args, **kwargs)
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/notebook/notebook/handlers.py", line 59, in get
> get_custom_frontend_exporters=get_custom_frontend_exporters
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/notebook/base/handlers.py", line 424, in render_template
> return template.render(**ns)
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/jinja2/environment.py", line 1008, in render
> return self.environment.handle_exception(exc_info, True)
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/jinja2/environment.py", line 780, in handle_exception
> reraise(exc_type, exc_value, tb)
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/jinja2/_compat.py", line 37, in reraise
> raise value.with_traceback(tb)
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/notebook/templates/notebook.html", line 1, in top-level template code
> {% extends "page.html" %}
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/notebook/templates/page.html", line 154, in top-level template code
> {% block header %}
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/notebook/templates/notebook.html", line 117, in block "header"
> {% for exporter in get_custom_frontend_exporters() %}
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/notebook/notebook/handlers.py", line 24, in get_custom_frontend_exporters
> exporter = get_exporter(name)()
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/nbconvert/exporters/base.py", line 95, in get_exporter
> return entrypoints.get_single('nbconvert.exporters', name).load()
> File "/Users/elliottcollins/anaconda/lib/python3.5/site-packages/entrypoints.py", line 81, in load
> obj = getattr(obj, attr)
> AttributeError: module 'hide_code' has no attribute 'HideCodeLatexPDFExporter'
> [E 13:15:42.382 NotebookApp] {
> "Connection": "keep-alive",
> "Accept-Language": "en-US,en;q=0.9",
> "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,/;q=0.8",
> "Referer": "http://localhost:8888/tree",
> "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/66.0.3359.181 Safari/537.36",
> "Host": "localhost:8888",
> "Cookie": "_xsrf=2|b23ab7ab|0b300fb5e6a8bc7867593fd0fd920127|1528143260; username-localhost-8888="2|1:0|10:1528143326|23:username-localhost-8888|44:YTU4MzZmMzU1ZjE3NGQ3YTgzYjVmZjBjMjdlYzdjYTE=|636e44c0aebff29c620d2a6461f86e67844173379895a325d628998abb741a29"",
> "Upgrade-Insecure-Requests": "1",
> "Accept-Encoding": "gzip, deflate, br"
> }
> [E 13:15:42.383 NotebookApp] 500 GET /notebooks/Impact%20Sort.ipynb (::1) 148.91ms referer=http://localhost:8888/tree

**Comment by GregSilverman** [comment]
Interesting @elliottmcollins. Mine works fine when hide_code is enabled (that is, my notebook open up in the correct state I had previously saved it in). I'm using Mac OS 10.12.6, Jupyter version 5.4.1 and nbconvert 5.3.1.

**Comment by elliottmcollins** [comment]
I'm using macOS 10.13.2, Jupyter 4.4.0, and nbconvert 5.3.1

Any chance you had to include the --ignore-installed option when installing
hide_code via pip?

**Comment by GregSilverman** [comment]
Nope, I did the install, verbatim, as outlined in the README.

**Comment by kirbs-** [comment]
@elliottmcollins 500 errors usually indicated the server extension isn't enabled. Verify with `jupyter serverextension list`; you should see the status in the output.

**Comment by kirbs-** [comment]
I think the warning you two are seeing is due to using the old jupyter_contrib_nbextensions. @GregSilverman I see the plugin listed in the output from `jupyter nbextension list`. @elliottmcollins do you have this plugin enabled too?

You can safely ignore the warning. It won't affect hide_code usage. It only means the web UI the extension provides won't work with hide_code.

**Comment by kirbs-** [comment]
@elliottmcollins the 404 you're seeing is because the export route is messed up. Try reopening the notebook (don't worry about restarting the kernel). This should reset all the exporting routes.
```
[W 14:04:03.424 NotebookApp] 404 GET /notebooks/development/python/mb_analysis/mb_latest.ipynb/export/pdf (::1): No such file or directory: development/python/mb_analysis/mb_latest.ipynb/export/pdf
[W 14:04:03.425 NotebookApp] 404 GET /notebooks/development/python/mb_analysis/mb_latest.ipynb/export/pdf (::1) 1.26ms referer=http://localhost:8888/notebooks/development/python/mb_analysis/mb_latest.ipynb#

```
You can also try opening http://localhost:8888/notebooks/development/python/mb_analysis/mb_latest.ipynb/export/pdf and see if it exports a PDF correctly.

**Comment by GregSilverman** [comment]
@kirbs-,
I have reopened and restarted Jupyter a whole bunch of times and I keep getting the 404 error. The URL you gave also does that. I am content with doing this from the command line, so no rush.

**Comment by kirbs-** [comment]
@GregSilverman 404 errors usually boil down to jupyter can't find the ipynb file. Is development/python/mb_analysis/mb_latest.ipynb a valid path from the root jupyter directory. i.e. can you navigate to the file from the directory view when jupyter notebook first launches?

Also checkout issue #18. I keep forgetting to add this quirk in configuration to the wiki.

**Comment by GregSilverman** [comment]
Yes ~/development/python/mb_analysis/mb_latest.ipynb is a valid path. Issue #18 however looks like it could potentially apply to my situation, since the absolute path is /Users/gregsilverman/development/python/mb_analysis/... again, I am not too concerned, since I have a workable solution by doing this at the command line.

**Comment by ANazaret** [comment]
I had the same error. I found a way to avoid the error but not to fix it (the PDF export won't work).
Since module 'hide_code' has no attribute 'HideCodeLatexPDFExporter', just add it.

Go to the "hide_code" package installation folder 
 - for me it's /usr/local/lib/python3.5/dist-packages/hide_code
 - for you it seems to be /home/dans/projects/Daily_Instrument_Review/DIR/lib/python3.6/site-packages/hide_code

Edit (with root privileges) the file hide_code.py to add the missing attribute.
I did it by replacing line 10 with: 
from .hide_code_pdf_exporter import HideCodePDFExporter as HideCodeLatexPDFExporter

Save your changes and reboot Jupyter !
